query/src/storages/fuse/operations/append.rs

```diff
@@ -116,15 +113,21 @@ impl FuseTable {
             )
         })?;
 
+        let mut cluster_keys_index = Vec::with_capacity(self.order_keys.len());
         if !self.order_keys.is_empty() {
             let input_schema = self.table_info.schema();
             let mut merged = input_schema.fields().clone();
 
             for expr in &self.order_keys {
                 let cname = expr.column_name();
-                if !merged.iter().any(|x| x.name() == &cname) {
-                    merged.push(expr.to_data_field(&input_schema)?);
-                }
+                let index = merged
+                    .iter()
+                    .position(|x| x.name() == &cname)
+                    .unwrap_or_else(|| {
+                        merged.push(expr.to_data_field(&input_schema).unwrap());
```

Comment:
seems that `unwrap` can be avoided here ... (at least if fall back to imperative style)